src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  handleClose?: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo: { title, id, user, completed },
```

Comment:
According to the checklist, you should avoid initializing state with props. This can lead to bugs and unexpected behavior. Instead, consider passing the props directly to the component that needs them, or using them in a useEffect hook if necessary.

Reply:
why in the hell do i need to useEffect instead of props and how it's gonna work?